node/src/components/block_synchronizer/block_builder.rs

```diff
@@ -456,6 +461,23 @@ impl BlockBuilder {
         }
     }
 
+    pub(super) fn waiting_for_block_header(&self) -> bool {
+        match &self.acquisition_state {
+            BlockAcquisitionState::Initialized(..) => true,
```

Comment:
an if let for these methods would probably be less annoying.

Reply:
I've given some thought and while I agree it's annoying using a wildcard will be less helpful if we refactor this logic in the future to add/remove states. Having a match here would help guide future changes (at least force us to reason about it) so I would prefer if we leave it as is.